### ISSUE TITLE
Lint for expect_true(all(x == y))

### ISSUE THREAD

**Issue by MichaelChirico** [jimhester/lintr#2927]
I guess this should be subsumed into some existing test.

Actually this one is quite pernicious, observe:

```r
x = NULL
y = 1.0

expect_true(all(x == y))
# passes!!
expect_identical(x, y)
# target is NULL, current is numeric
```
